recipes/msgpack/all/conanfile.py

```diff
@@ -0,0 +1,82 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class MsgpackConan(ConanFile):
+    name = "msgpack"
+    description = "The official C++ library for MessagePack"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/msgpack/msgpack-c"
+    topics = ("conan", "msgpack", "message-pack", "serialization")
+    license = "BSL-1.0"
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    settings = "os", "arch", "build_type", "compiler"
+    options = {"fPIC": [True, False], "shared": [True, False], "header_only": [True, False], "with_boost": [True, False]}
+    default_options = {"fPIC": True, "shared": False, "header_only": False, "with_boost": False}
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.header_only:
+            self.settings.clear()
```

Comment:
Input settings should not be cleared. Why are you doing this?

Reply:
You only need to clear them in package_id() as you are already doing, so this line should be removed